pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -118,84 +118,113 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 		}
 	}
 
-	// manage status
-	var availableConditionReasons []string
+	flag := false
+	operatorConfig.Status.Conditions = nil
```

Comment:
Don't do this, you might be wiping out conditions from other control loops.
Clear/remove the "legacy" `Available` condition explicitly.

Reply:
Is it advisable to use the `RemoveOperatorCondition` method for clearing/removing conditions?
https://github.com/openshift/library-go/blob/451a0ed3a6aede5a64d95d2d4382c74e879f5e1f/pkg/operator/v1helpers/helpers.go#L66

Reply:
That looks good.